### DISCUSSION TITLE
Injections/Build/Add on/Field Conditions

### DISCUSSION THREAD

**Discussion by Tedoemoney** [riclolsen/OSHMI#10] [Q&A] [answered]
Hello Mr. Olsen. Thank you for your build and contribution for this product. I'm new to the IT part of the world, but old school when it comes to SCADA. I was wondering if there is a way to get behind the scenes and do some testing with injections. 

1. I would like to know if possible that scenarios could be built and then attack injections on the system? 
2. I was wondering if any of the behind-the-scenes data can be altered. How can I see the messages pass through the network and capture those packets (goose messages) so that we can study possible weaknesses in systems that are built.  Number of feeders, T/D banks, and lastly any additional IEDs that could be installed to get from Breaker/Feeder to installing branch line fusing and practice fuse coordination, loss of power rerouting, can any of that data be altered? 
3. Would like to see if we could go from Feeder to the meter of the home, and then from Generation to Transmission (top of transmission side of substation). Maybe building out full scale to match existing field conditions, (final goal) 

As I said, I'm extremely new to all of this and was wondering if this would be possible, I know this is a lot to ask (to say the least).  I appreciate your time and thought. Thank you!

**Comment by riclolsen** [comment, marked as answer]
Hello Ted,

2. Goose messages are currently not supported by the IEC61850 client driver, just MMS messages. You can use Wireshark to capture network packets. Unencrypted protocol messages can be altered, yes.

1./3. OSHMI does not work as a gateway, so you need some other software stack to build this feeder-to-meter scenario. I think you can build it with some free 61850 demo server and an open source gateway (maybe JSON-SCADA or IndigoScada).

Best regards,
Ricardo
